src/Std/Time.lean

```diff
@@ -0,0 +1,174 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+import Std.Time.Time
+import Std.Time.Date
+import Std.Time.Zoned
+import Std.Time.Format
+import Std.Time.DateTime
+import Std.Time.Notation
+
+namespace Time
+
+/-!
+# Time
+
+The Lean4 API for date, time, and duration functionalities, following the ISO standards.
+
+# Overview
+
+This module of the standard library defines various concepts related to time, such as dates, times,
+time zones and durations. These types are designed to be strongly-typed and to avoid problems with
+conversion. It offers both unbounded and bounded variants to suit different use cases, like
+adding days to a date or representing ordinal values.
+
+# Date and Time Components
+
+Date and time components are classified into different types based how you SHOULD use them. These
+components are categorized as:
+
+## Offset
```

Comment:
I think it is very important to have such a concept of time offsets that are meaningful to humans. I'm just wondering whether it would be more ergonomic if all of these were bundled into a type like Java's [Period](https://docs.oracle.com/en%2Fjava%2Fjavase%2F21%2Fdocs%2Fapi%2F%2F/java.base/java/time/Period.html).

Reply:
I'm not sure if it's the right approach, `Period` only stores YMD, these types are useful to represent units of time and convert between them easily like the `o-clock` library from Haskell.